src/datastore/postgres-store.ts

```diff
@@ -3401,6 +3437,147 @@ export class PgDataStore
     });
   }
 
+  getTxListEvents(args: {
+    txs: {
+      txId: string;
+      indexBlockHash: string;
+    }[];
+    limit: number;
+    offset: number;
+  }) {
+    return this.queryTx(async client => {
+      // preparing condition to query from
+      // condition = (tx_id=$1 AND index_block_hash=$2) OR (tx_id=$3 AND index_block_hash=$4)
+      // let condition = this.generateParameterizedWhereAndOrClause(args.txs);
+      if (args.txs.length === 0) return { results: [] };
+      let condition = '(tx_id, index_block_hash) = ANY(VALUES ';
+      let counter = 1;
+      const transactionValues = args.txs
+        .map(_ => {
+          const singleCondition = '($' + counter + '::bytea, $' + (counter + 1) + '::bytea)';
+          counter += 2;
+          return singleCondition;
+        })
+        .join(', ');
+      condition += transactionValues + ')';
```

Comment:
Thanks @MHassanTariq for the readability improvements in this query!

At first glance this seems okay performance-wise, but we need to do some manually testing against the mainnet dataset to be sure. We may need to add a composite index for `(tx_id, index_block_hash)`.

Reply:
cc @asimm241 is the performance profiling something you could help with?

Also, if you can come up with an isolated SQL query with some sample mainnet data, I can test against our prod DBs.